src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEmbeddedAttributeSymbol.cs

```diff
@@ -0,0 +1,171 @@
+
+        public override ImmutableArray<Symbol> GetMembers() => _members;
+
+        public override ImmutableArray<Symbol> GetMembers(string name) => name == WellKnownMemberNames.InstanceConstructorName ? _members : ImmutableArray<Symbol>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers() => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int arity) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override ImmutableArray<string> GetAppliedConditionalSymbols() => ImmutableArray<string>.Empty;
+
+        internal override AttributeUsageInfo GetAttributeUsageInfo() => AttributeUsageInfo.Default;
+
+        internal override NamedTypeSymbol GetDeclaredBaseType(ConsList<Symbol> basesBeingResolved) => _baseType;
+
+        internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList<Symbol> basesBeingResolved) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers() => GetMembers();
+
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers(string name) => GetMembers(name);
+
+        internal override IEnumerable<FieldSymbol> GetFieldsToEmit() => SpecializedCollections.EmptyEnumerable<FieldSymbol>();
+
+        internal override ImmutableArray<NamedTypeSymbol> GetInterfacesToEmit() => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override IEnumerable<SecurityAttribute> GetSecurityInformation() => null;
+
+        internal override ImmutableArray<NamedTypeSymbol> InterfacesNoUseSiteDiagnostics(ConsList<Symbol> basesBeingResolved = null) => ImmutableArray<NamedTypeSymbol>.Empty;
+    }
+
+    internal sealed class SynthesizedEmbeddedAttributeConstructorSymbol : SynthesizedInstanceConstructor
+    {
+        private readonly ImmutableArray<ParameterSymbol> _parameters;
+
+        internal SynthesizedEmbeddedAttributeConstructorSymbol(
+            NamedTypeSymbol containingType,
+            Func<MethodSymbol, ImmutableArray<ParameterSymbol>> getParameters) :
+            base(containingType)
+        {
+            _parameters = getParameters(this);
+        }
+
+        public override ImmutableArray<ParameterSymbol> Parameters => _parameters;
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            var factory = new SyntheticBoundNodeFactory(this, this.GetNonNullSyntaxNode(), compilationState, diagnostics);
+            factory.CurrentMethod = this;
+
+            var baseType = ContainingType.BaseType;
+            var baseConstructor = baseType.InstanceConstructors.SingleOrDefault(m => m.ParameterCount == 0);
```

Comment:
>SingleOrDefault [](start = 64, length = 15)

Is it going to throw in case of an ambiguous match?

Reply:
Changed to `FirstOrDefault`.